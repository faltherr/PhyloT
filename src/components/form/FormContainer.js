import React, {Component} from 'react'
import {connect} from 'react-redux'
import { Link } from 'react-router-dom'

//Local Components
import PlatformSelector from './PlatformSelector'

//Styles
import '../../styles/mainwizard.css'


//This component holds the functionality of the main form page. It may be converted to a wizard like feature in the future.
class FormContainer extends Component{
    constructor(){
        super()
        this.state={
            currentStep: 1
        }
    }

    displayButton = () =>{
        if(this.props.inputType === 'existingProject' || this.props.inputType === 'biom'){
            return <Link to = '/generate/review'><button type='button' className='btn btn-success'>Review Sample</button></Link>
        } else if (this.props.inputType === 'newProject'){
            return <Link to='/generate/customize'> <button type='button' className='btn btn-success'>Select Genomes for Sample</button> </Link>
        } else{
            return null
        }
    }

    render(){
        return(
            <div className='wizard-main-container'>
            <section className='top-text'>
                <h1>Genome Selector</h1>
                <p>Use seqSim to customize a synthetic genome community or upload a file of your own.</p>
            </section>
            <div className='main-content'>
                <div className='wizard'>
                    <div className='content-container'>
                        
                        <PlatformSelector/>
                    </div>
                    <div className='button-container'>
                        {this.displayButton()}
                    </div>
                </div>
            </div>
            </div>
        )
    }
}

let mapStateToProps = state => {
    return(
        state
    )
}

export default connect(mapStateToProps)(FormContainer)