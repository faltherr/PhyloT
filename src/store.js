import {createStore, applyMiddleware} from 'redux'
import promiseMiddleware from 'redux-promise-middleware'

import mainReducer from './reducers/mainReducer'

const middleware = [promiseMiddleware]

export default createStore( mainReducer, applyMiddleware(...middleware))