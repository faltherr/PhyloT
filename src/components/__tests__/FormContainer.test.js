import { configure, shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import store from '../../store'
import FormContainer from '../form/FormContainer'

configure({ adapter: new Adapter() })

describe('FormContainer', () => {
    //On the first run of this test Jest generates a snapshot that will be compared in future tests
    //If the component is intentionally changed we should update the snapshot `jest --updateSnapshot`
    it('Renders to DOM', () => {
        const wrapper = shallow(<FormContainer store={store}/>);
        expect(toJSON(wrapper)).toMatchSnapshot()
    })
    // it ('Renders two components', ()=>{})
})