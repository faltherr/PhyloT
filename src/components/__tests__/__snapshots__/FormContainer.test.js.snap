// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormContainer Renders to DOM 1`] = `
<FormContainer
  biomFile={Array []}
  collection={Array []}
  customModel={Array []}
  dispatch={[Function]}
  existingProjectFile={Array []}
  genomeSample={Array []}
  inputType=""
  radioInputClassChecked={false}
  readModel="default"
  selectedGenome={Array []}
  seqPlatform="illumina"
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
  taxIdSearch={Array []}
/>
`;
